### DISCUSSION TITLE
MHI2_ER_POG11_K5158 1443 - No Patch file?

### DISCUSSION THREAD

**Discussion by NMarq** [Mr-MIBonk/M.I.B._More-Incredible-Bash#86] [Q&A] [answered]
Hi,
We have software version MHI2_ER_POG11_K5158 1443 which does not seem to have a patch file available? 

Can we use MHI2_ER_POG11_P5158 1443 or does a new patch need creating? 

Can anyone help/advise.

Many thanks
Nick

**Comment by harman-f** [comment]
Yes, you can rename the folder.

**Reply by NMarq** [reply]
Thank you.

**Reply by NMarq** [reply]
This unfortunately does not seem to work. The system installs the MIB Solution without any issues, but after patching and the enabling processes for CarPlay/AA etc, nothing is enabled. No AA, no ability to update later map set etc.

Can anyone offer any help? 

Thanks.

**Reply by harman-f** [reply]
please share your log file

**Comment by NMarq** [comment]
Here you go:

https://www.dropbox.com/s/75ucop2f00hg26p/MU1443-JTB-07426.04.1811760778.log?dl=0

Had a quick look at the backup IFS file, the HEX EB 7E FF does not seem to start at 00BA0000 but then I've never looked at this stuff before, so who knows..... Scrap that, looking at the wrong file.... 

Thank you :)

**Reply by harman-f** [reply]
```


 __  __     ___           ___  
|  \/  |   |_ _|         | _ ) 
| |\/| |    | |          | _ \ 
|_|  |_|ore|___|ncredible|___/ash - V3.4.4 "Ambient MOD..."
______________________________________________________________________

NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF

Logging_MU1443-MHI2_ER_POG11_K5158-JTB-********************:
HW Number: 054 - PN: 9P1035036L
Variant: FM2-P-TNL-EU-PO-MLP
EEPROM Variant: MU-P-LN-EU
Dataset Version: -----------
Model ID: 074425 - Car Model: Porsche 718 Boxster
VIN -> CAR:  - MIB: 

 ls --> total 2919
      64 drwxrwxrwx 11 root      root          32768 Jan 01 00:00 .
      64 drwxrwxrwx 11 root      root          32768 Jan 01 00:00 ..
      64 drwxrwxrwx  2 root      root          32768 Jan 05  2023 .System Volume Information
       2 -rwxrwxrwx  1 root      root            986 Dec 19  2022 .git-blame-ignore-revs
       1 -rwxrwxrwx  1 root      root            399 Dec 19  2022 .gitattributes
      64 drwxrwxrwx  2 root      root          32768 Dec 19  2022 .github
       1 -rwxrwxrwx  1 root      root            132 Dec 19  2022 .gitignore
    1703 -rwxrwxrwx  1 root      root         871810 Dec 19  2022 2022_06_MHI2_Password_List_V4.0.pdf
      81 -rwxrwxrwx  1 root      root          41255 Dec 19  2022 GEM.png
      30 -rwxrwxrwx  1 root      root          14939 Dec 19  2022 LICENSE
      64 drwxrwxrwx  4 root      root          32768 Dec 19  2022 Launcher
       1 -rwxrwxrwx  1 root      root             60 Dec 19  2022 M.I.B. - More Incredible Bash - Mibsolution Wiki.url
       0 -rwxrwxrwx  1 root      root              0 Dec 19  2022 NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF.txt
     284 -rwxrwxrwx  1 root      root         144962 Dec 19  2022 PATCH COMPATIBILITY TABLE.pdf
      13 -rwxrwxrwx  1 root      root           6469 Dec 19  2022 README.md
       1 -rwxrwxrwx  1 root      root             24 Dec 19  2022 VERSION
       1 -rwxrwxrwx  1 root      root            296 Jan 01  1980 WMPInfo.xml
       0 -rwxrwxrwx  1 root      root              0 Dec 19  2022 _Swdlautorun.txt
      64 drwxrwxrwx  3 root      root          32768 Dec 19  2022 apps
      64 drwxrwxrwx  4 root      root          32768 Jan 01  1980 backup
      77 -rwxrwxrwx  1 root      root          39351 Dec 19  2022 bash.png
      64 drwxrwxrwx  2 root      root          32768 Dec 19  2022 config
      64 drwxrwxrwx  3 root      root          32768 Dec 19  2022 esd
       4 -rwxrwxrwx  1 root      root           1995 Dec 19  2022 metainfo2.txt
      64 drwxrwxrwx 10 root      root          32768 Dec 19  2022 mod
      64 drwxrwxrwx  4 root      root          32768 Dec 19  2022 patches
      16 -rwxrwxrwx  1 root      root           7874 Dec 19  2022 start

RCC-backup---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Version info..... -> done :-)
FEC folders...... -> done :-)
Shadow files..... -> done :-)
Dataset's........ -> done :-)
EEPROM........... -> done :-)
storage1......... -> done :-)
storage2......... -> done :-)
backing up backup -> part 1 done
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 MU1443
Free space on /net/mmx/mnt/boardbook 893M


RCC-launcher---->
-- _Swdlautorun.txt found
-- autostart already disabled
-- Skip MOST enabled
-- Illegal and withdrawn FECs removed
-- Cleanup done!
AppDevelopment.jar found
-- GEM version OK - no update needed
-- M.I.B GEM link installed
-- Development Mode - GEM - activated


RCC-backup---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Version info..... -> Backup is already there!
FEC folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
backing up backup -> Backup part 1 is already there!
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 MU1443
Free space on /net/mmx/mnt/boardbook 893M


RCC-launcher---->
-- _Swdlautorun.txt found
-- autostart already disabled
-- Skip MOST enabled
-- Illegal and withdrawn FECs removed
-- Cleanup done!
AppDevelopment.jar found
-- GEM version OK - no update needed
-- M.I.B GEM link installed
-- Development Mode - GEM - activated

MMX-navon---->
Navigation is activated now
000000010000000003111102060800132f0037e6037041069e
Reboot Unit...

RCC-carp---->
CarPlay and Android Auto activated now

RCC-addimage---->
File checksum does not match. No patch applied
Button images are already installed

RCC-flash---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Partition info... -> Backup is already there!
SWDL............. -> Backup is already there!
Version info..... -> Backup is already there!
GEM info........ -> done :-)
Folder lists..... -> Backup is already there!
FEC folders...... -> Backup is already there!
HMI folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
lsd.jxe.......... -> Backup is already there!
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
RCC Flashlock.... -> Backup is already there!
RCC.............. -> Backup is already there!
MMX.............. -> Backup is already there!
IFS-stage2.......4512+0 records in
4512+0 records out
 -> done :-)
OFFSET's......... -> done :-)
backing up backup -> Backup part 1 is already there!
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:05 MU1443
Free space on /net/mmx/mnt/boardbook 803M


search File MU1443-ifs-root-part2-0x00ba0000-4006F400.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00f4064c (15992396)
End addr:   0x01ae064c
*** erasing:     0x00BA0000 .. 0x01AE064B [01AFFFFF] ........................................................................................................................... 
*** programming: 0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
*** verifying:   0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
[flashlock] locking flash
0 image type=1(IPL     )       0...  1ffff (  131072)|0...1 (1):LOCKED
1 image type=2(IFS     )   20000... 3b0e53 ( 3739220)|1...30 (29):LOCKED
2 image type=2(IFS     )  540000... b9dc07 ( 6675464)|42...93 (51):LOCKED
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
4 image type=4(DSP     ) 3d00000...3dc6923 (  813348)|488...495 (7):LOCKED
5 image type=5(BIOS/ADJ) 3f00000...3ffffff ( 1048576)|504...512 (8):NOT LOCKED
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
0x00ba0000 - 4C06F400

dump RCC to logs, that takes some seconds......... -> done :-)

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
copy EL file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
Reboot Unit...

MMX-carp---->
CarPlay and Android Auto activated now
Reboot Unit...

RCC-addfec---->
FECs from /net/rcc/mnt/efs-persist/FEC
FECs currently used by unit
00060900
00060500
00070200
00070100
00040100
06300099
00030000
00050000
00060700
00060800
00060100
00060A00

FECs from /net/rcc/mnt/efs-system/backup/FEC
FECs from factory
00060500
00070200
00070100
00040100
06300099
00030000
00050000
00060700
00060800
00060100
00060A00


MMX-fecel---->

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)

MMX-fecel---->
copy EL file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)

RCC-addfec---->
FECs from /net/rcc/mnt/efs-persist/FEC
FECs currently used by unit
00060900
00060500
00070200
00070100
00040100
06300099
00030000
00050000
00060700
00060800
00060100
00060A00

FECs from /net/rcc/mnt/efs-system/backup/FEC
FECs from factory
00060500
00070200
00070100
00040100
06300099
00030000
00050000
00060700
00060800
00060100
00060A00


RCC-NavActiveIgnore---->
no supported train found - will stop here

RCC-svm---->
Fix SVM Error (Note: The process can take some minutes...)
extract build version...
SvmInfo - Compressing... /net/rcc/mnt/efs-persist/SWDL/Log/SvmInfo.bin CfgAckRand - Uncompressing...
 /net/rcc/mnt/efs-persist/SWDL/Log/CfgAckRand.z to /net/rcc/mnt/efs-persist/SWDL/Log/CfgAckRand.bin

Key: 706D - CHKSum: B9BF

CfgAck - Compressing... /net/rcc/mnt/efs-persist/SWDL/Log/CfgAck.bin Wait for DSI to connect ...
DSI Connected role1.DSIE2PAccess
responseWriteE2P address 0x3f0 status 0 len 0x03
0x3f0 	00 00 01 
DSI Disconnected
clear SVM done!

Reboot Unit...
---- Read original Variant_1 from unit
Wait for DSI to connect ...
DSI Connected role1.DSIE2PAccess
0xba 	4d 55 2d 50 2d 4c 4e 2d 45 55 00 00 00 
DSI Disconnected
---- Clean Variant - 13 bytes 00
Wait for DSI to connect ...
DSI Connected role1.DSIE2PAccess
responseWriteE2P address 0xba status 0 len 0x0d
0xba 	00 00 00 00 00 00 00 00 00 00 00 00 00 
DSI Disconnected
---- Write Variant_1
Wait for DSI to connect ...
DSI Connected role1.DSIE2PAccess
responseWriteE2P address 0xba status 0 len 0x0a
0xba 	4d 55 2d 50 2d 4c 4e 2d 45 55 
DSI Disconnected
---- Read Variant_1 back for log
Wait for DSI to connect ...
DSI Connected role1.DSIE2PAccess
0xba 	4d 55 2d 50 2d 4c 4e 2d 45 55 00 00 00 
DSI Disconnected

MMX-navon---->
Navigation is activated now
000000010000000003111102060800132f0037e6037041069e
Reboot Unit...
Reboot Unit...

MMX-carp---->
CarPlay and Android Auto activated now
Reboot Unit...


```

**Reply by harman-f** [reply]
Patch did run fine.
```
search File MU1443-ifs-root-part2-0x00ba0000-4006F400.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00f4064c (15992396)
End addr:   0x01ae064c
*** erasing:     0x00BA0000 .. 0x01AE064B [01AFFFFF] ........................................................................................................................... 
*** programming: 0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
*** verifying:   0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
[flashlock] locking flash
0 image type=1(IPL     )       0...  1ffff (  131072)|0...1 (1):LOCKED
1 image type=2(IFS     )   20000... 3b0e53 ( 3739220)|1...30 (29):LOCKED
2 image type=2(IFS     )  540000... b9dc07 ( 6675464)|42...93 (51):LOCKED
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
4 image type=4(DSP     ) 3d00000...3dc6923 (  813348)|488...495 (7):LOCKED
5 image type=5(BIOS/ADJ) 3f00000...3ffffff ( 1048576)|504...512 (8):NOT LOCKED
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
0x00ba0000 - 4C06F400

dump RCC to logs, that takes some seconds......... -> done :-)

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
copy EL file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
Reboot Unit...
```

Fec `00060900` was added to unit.

All FECs are there

Btw. you have to connect a phone to see the AA/CP button.

**Reply by NMarq** [reply]
Hi,

Yes have connected a phone, CarPlay works (but it did from OEM) but AA does not. Map updates also do not work using VW Discover Pro maps updates which do work on another patched unit.

Thanks
Nick

**Reply by NMarq** [reply]
This is a bench unit I'm testing pre-fitting to a car BTW.

Assume that the lack of can bus traffic and VIN identification makes no difference?

Thanks
Nick

**Comment by harman-f** [comment]
For AA try a different USB cable.
For maps - what error do you get? - formatting navdb most likely helps.

Bench or not bench does not matter for this

**Comment by NMarq** [comment, marked as answer]
SOLUTION:
So, this is sorted and it was all because the unit was on the bench and not in car.

More specifically:

Android Auto - Needs Canbus data to the head unit before it will function, without it the USB just detects and charges only. No idea why this is especially as CarPlay works fine on the bench.

Map updates - This one confused me at first as it didn't work in car; I'd only connected the main quad lock and display for testing, but then I thought perhaps the map update needs a valid GPS signal before it will update. That is indeed the case, the map update will only work with a valid GPS signal. Again very strange, perhaps it is ensuring the unit is in Europe before it allows a European map update.

Thanks again for all your help
Nick


Original Question:
Hi, 
A replacement cable doesn't make a difference, I'm going to try another phone (even thou the same phone does work with AA on the other unit.)

The map update is giving 'The medium contains no valid update data'. Again this same SD card was used for update on the other unit.

For info, the other unit was updated using 'proper' FEC codes using a different tool.

Any ideas? It is most odd.

I also have another unit with a different firmware, so will also try that over the next few days as well and post back here.

Thanks again
Nick

**Reply by harman-f** [reply]
This is very strange.

Your issues suggest thst the patched ifs-root was not applied, but the log looks fine.
At the same place as the log file you should also find a *ifs file. get this please.
This was dumped from unit AFTER patching, hence will show if the patch ended up on the unit.

Was navigation enbaled on this unit before?

**Reply by NMarq** [reply]
Please see link to the IFS file and below a new log dump for consistency as we are uninstalled and re-installed since the above.

Yes navigation was enabled and working pre-patch.

IFS Dump - https://www.dropbox.com/s/oho5q7n96vgrj3t/MU1443-JTB-07426.04.1811760778-ifs-root-part2-0x00ba0000-4C06F400.rcc?dl=0 


 __  __     ___           ___  
|  \/  |   |_ _|         | _ ) 
| |\/| |    | |          | _ \ 
|_|  |_|ore|___|ncredible|___/ash - V3.4.4 "Ambient MOD..."
______________________________________________________________________

NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF

Logging_MU1443-MHI2_ER_POG11_K5158-JTB-07426.04.1811760778:
HW Number: 054 - PN: 9P1035036L
Variant: FM2-P-TNL-EU-PO-MLP
EEPROM Variant: MU-P-LN-EU
Dataset Version: -----------
Model ID: 000000 - Car Model: not set
VIN -> CAR:  - MIB: 

 ls --> total 2919
      64 drwxrwxrwx 11 root      root          32768 Jan 01 00:00 .
      64 drwxrwxrwx 11 root      root          32768 Jan 01 00:00 ..
      64 drwxrwxrwx  2 root      root          32768 Jan 16  2023 .System Volume Information
       2 -rwxrwxrwx  1 root      root            986 Dec 19  2022 .git-blame-ignore-revs
       1 -rwxrwxrwx  1 root      root            399 Dec 19  2022 .gitattributes
      64 drwxrwxrwx  2 root      root          32768 Dec 19  2022 .github
       1 -rwxrwxrwx  1 root      root            132 Dec 19  2022 .gitignore
    1703 -rwxrwxrwx  1 root      root         871810 Dec 19  2022 2022_06_MHI2_Password_List_V4.0.pdf
      81 -rwxrwxrwx  1 root      root          41255 Dec 19  2022 GEM.png
      30 -rwxrwxrwx  1 root      root          14939 Dec 19  2022 LICENSE
      64 drwxrwxrwx  4 root      root          32768 Dec 19  2022 Launcher
       1 -rwxrwxrwx  1 root      root             60 Dec 19  2022 M.I.B. - More Incredible Bash - Mibsolution Wiki.url
       0 -rwxrwxrwx  1 root      root              0 Dec 19  2022 NOT FOR COMMERCIAL USE - IF YOU BOUGHT THIS YOU GOT RIPPED OFF.txt
     284 -rwxrwxrwx  1 root      root         144962 Dec 19  2022 PATCH COMPATIBILITY TABLE.pdf
      13 -rwxrwxrwx  1 root      root           6469 Dec 19  2022 README.md
       1 -rwxrwxrwx  1 root      root             24 Dec 19  2022 VERSION
       1 -rwxrwxrwx  1 root      root            296 Jan 01  1980 WMPInfo.xml
       0 -rwxrwxrwx  1 root      root              0 Dec 19  2022 _Swdlautorun.txt
      64 drwxrwxrwx  3 root      root          32768 Dec 19  2022 apps
      64 drwxrwxrwx  4 root      root          32768 Jan 01  1980 backup
      77 -rwxrwxrwx  1 root      root          39351 Dec 19  2022 bash.png
      64 drwxrwxrwx  2 root      root          32768 Dec 19  2022 config
      64 drwxrwxrwx  3 root      root          32768 Dec 19  2022 esd
       4 -rwxrwxrwx  1 root      root           1995 Dec 19  2022 metainfo2.txt
      64 drwxrwxrwx 10 root      root          32768 Dec 19  2022 mod
      64 drwxrwxrwx  4 root      root          32768 Dec 19  2022 patches
      16 -rwxrwxrwx  1 root      root           7874 Dec 19  2022 start

RCC-backup---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Version info..... -> done :-)
FEC folders...... -> done :-)
Shadow files..... -> done :-)
Dataset's........ -> done :-)
EEPROM........... -> done :-)
storage1......... -> done :-)
storage2......... -> done :-)
backing up backup -> Backup part 1 is already there!
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:05 MU1443
Free space on /net/mmx/mnt/boardbook 803M


RCC-launcher---->
-- _Swdlautorun.txt found
-- autostart already disabled
-- Skip MOST enabled
-- Illegal and withdrawn FECs removed
-- Cleanup done!
AppDevelopment.jar found
-- GEM version OK - no update needed
-- M.I.B GEM link installed
-- Development Mode - GEM - activated

RCC-carp---->
CarPlay and Android Auto activated now

RCC-addimage---->
File checksum does not match. No patch applied
Button images are already installed

RCC-flash---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Partition info... -> done :-)
SWDL............. -> done :-)
Version info..... -> Backup is already there!
GEM info........ -> done :-)
Folder lists..... -> done :-)
FEC folders...... -> Backup is already there!
HMI folders...... -> done :-)
Shadow files..... -> Backup is already there!
lsd.jxe.......... -> done :-)
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
RCC Flashlock.... -> done :-)
RCC.............. -> done :-)
MMX.............. -> done :-)
IFS-stage2.......4512+0 records in
4512+0 records out
 -> done :-)
OFFSET's......... -> done :-)
backing up backup -> Backup part 1 is already there! -> Backup part 2 is already there!

ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:05 MU1443
Free space on /net/mmx/mnt/boardbook 803M


search File MU1443-ifs-root-part2-0x00ba0000-4006F400.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00f4064c (15992396)
End addr:   0x01ae064c
*** erasing:     0x00BA0000 .. 0x01AE064B [01AFFFFF] ........................................................................................................................... 
*** programming: 0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
*** verifying:   0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
[flashlock] locking flash
0 image type=1(IPL     )       0...  1ffff (  131072)|0...1 (1):LOCKED
1 image type=2(IFS     )   20000... 3b0e53 ( 3739220)|1...30 (29):LOCKED
2 image type=2(IFS     )  540000... b9dc07 ( 6675464)|42...93 (51):LOCKED
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
4 image type=4(DSP     ) 3d00000...3dc6923 (  813348)|488...495 (7):LOCKED
5 image type=5(BIOS/ADJ) 3f00000...3ffffff ( 1048576)|504...512 (8):NOT LOCKED
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
0x00ba0000 - 4C06F400

dump RCC to logs, that takes some seconds......... -> done :-)

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
copy EL file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
Reboot Unit...

MMX-reboot---->
Reboot Unit...

RCC-flash---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
0x00ba0000 - 4C06F400

search JTB-07426.04.1811760778 Backup File MU1443-ifs-root-part2-0x00ba0000.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x011a0000 (18481152)
End addr:   0x01d40000
*** erasing:     0x00BA0000 .. 0x01D3FFFF [01D3FFFF] ............................................................................................................................................. 
*** programming: 0x00BA0000 .. 0x01D3FFFF            ............................................................................................................................................. 
*** verifying:   0x00BA0000 .. 0x01D3FFFF            ............................................................................................................................................. 
[flashlock] locking flash
0 image type=1(IPL     )       0...  1ffff (  131072)|0...1 (1):LOCKED
1 image type=2(IFS     )   20000... 3b0e53 ( 3739220)|1...30 (29):LOCKED
2 image type=2(IFS     )  540000... b9dc07 ( 6675464)|42...93 (51):LOCKED
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
4 image type=4(DSP     ) 3d00000...3dc6923 (  813348)|488...495 (7):LOCKED
5 image type=5(BIOS/ADJ) 3f00000...3ffffff ( 1048576)|504...512 (8):NOT LOCKED
copy EL file from backup JTB-07426.04.1811760778... -> done :-)

copy FEC file from backup JTB-07426.04.1811760778... -> done :-)

Reboot Unit...

RCC-backup---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Version info..... -> Backup is already there!
FEC folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
backing up backup -> Backup part 1 is already there!
ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:05 MU1443
Free space on /net/mmx/mnt/boardbook 803M


RCC-launcher---->
-- _Swdlautorun.txt found
-- autostart already disabled
-- Skip MOST enabled
-- Illegal and withdrawn FECs removed
-- Cleanup done!
AppDevelopment.jar found
-- GEM version OK - no update needed
-- M.I.B GEM link installed
-- Development Mode - GEM - activated

RCC-carp---->
CarPlay and Android Auto activated now

RCC-addimage---->
File checksum does not match. No patch applied
Button images are already installed

RCC-flash---->
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae063f (15992384)|93...216 (123):LOCKED
0x00ba0000 - 4006F400

Found JTB-07426.04.1811760778 Backup, will check if all complete...
Partition info... -> Backup is already there!
SWDL............. -> Backup is already there!
Version info..... -> Backup is already there!
GEM info........ -> done :-)
Folder lists..... -> Backup is already there!
FEC folders...... -> Backup is already there!
HMI folders...... -> Backup is already there!
Shadow files..... -> Backup is already there!
lsd.jxe.......... -> Backup is already there!
Dataset's........ -> Backup is already there!
EEPROM........... -> Backup is already there!
storage1......... -> Backup is already there!
storage2......... -> Backup is already there!
RCC Flashlock.... -> Backup is already there!
RCC.............. -> Backup is already there!
MMX.............. -> Backup is already there!
IFS-stage2....... -> Backup is already there!
OFFSET's......... -> Backup is already there!
backing up backup -> Backup part 1 is already there! -> Backup part 2 is already there!

ls -als /net/mmx/mnt/boardbook --> total 192
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:02 .
      64 drwxrwxr-x  7 root      root          32768 Jan 01 00:02 ..
      64 drwxrwxrwx  3 root      root          32768 Jan 01 00:05 MU1443
Free space on /net/mmx/mnt/boardbook 803M


search File MU1443-ifs-root-part2-0x00ba0000-4006F400.ifs

[flashlock] unlocking flash: DONEFlashing on OFFSET 0x00ba0000...
Label: PL_qnx_flashit_10495A QNX: RL_qnx_650_10494A Built: 2010/12/10_11-21-47_UTC
/net/rcc/dev/fs0
--------
Flash size: 0x04000000 [= 512 x 0x20000]
Base addr:  0x00ba0000
File size:  0x00f4064c (15992396)
End addr:   0x01ae064c
*** erasing:     0x00BA0000 .. 0x01AE064B [01AFFFFF] ........................................................................................................................... 
*** programming: 0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
*** verifying:   0x00BA0000 .. 0x01AE064B            ........................................................................................................................... 
[flashlock] locking flash
0 image type=1(IPL     )       0...  1ffff (  131072)|0...1 (1):LOCKED
1 image type=2(IFS     )   20000... 3b0e53 ( 3739220)|1...30 (29):LOCKED
2 image type=2(IFS     )  540000... b9dc07 ( 6675464)|42...93 (51):LOCKED
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
4 image type=4(DSP     ) 3d00000...3dc6923 (  813348)|488...495 (7):LOCKED
5 image type=5(BIOS/ADJ) 3f00000...3ffffff ( 1048576)|504...512 (8):NOT LOCKED
ifs-root-part2 (ASCII): 00f0
3 image type=2(IFS     )  ba0000...1ae064b (15992396)|93...216 (123):LOCKED
0x00ba0000 - 4C06F400

dump RCC to logs, that takes some seconds......... -> done :-)

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
copy EL file from /patches/MHI2_ER_POG11_K5158_MU1443_PATCH... -> done :-)
Reboot Unit...
Reboot Unit...

**Reply by harman-f** [reply]
You RCC dump is OK. The patched ifs-root-stage2 is on your unit.

This is odd.

**Reply by NMarq** [reply]
Yes indeed it is odd. If it was just the map update or just the Android Auto I would have put it down to my stupidity, but both not working is odd.
I'll test the other unit tomorrow that is one a different firmware version and post back here.

Thank you for all your help.

**Reply by NMarq** [reply]
So here is the update.

I have patched another unit that's a different firmware level and it's responds identically. I.e. Android Auto doesn't work.... My assumption is either I'm an idiot (definitely possible...) or the issue is related to the units being on the bench with no can bus data.

So for Android Auto - I see the head unit detect Android Auto and I get the acknowledgment prompt for Android Auto, but then it just starts charging the phone with no Android Auto prompts for set-up on the phone. The exact same phone with the exact same cable works without any issues in another in-car unit patched using a different method. (I've not progressed the map updates as we might as well resolve this issue first since they are both no doubt related.)

For clarity this is what I did for patching.

1. Copy the MIB Incredible Bash files to an SD card.
2. Copy the appropriate firmware patch folder/files to the patch folder on the SD card.
3. Run the MIB installation on the head unit.
4. Go into the green screen MIB.
5. Run the All in one patch.
6. Test Android Auto.

I not done anything with the FEC codes etc, etc, just left it as it comes.

Any ideas please?

I will over the next couple days put the head unit in car and see if AA works etc and post an update.

Thanks

**Reply by harman-f** [reply]
I'm currently also out of ideas.
Have seen many units working fine with AA. Can't remember anything was done differently.

Is CP working on the same car?

Otherweise, try additional different cables...

**Reply by NMarq** [reply]
So, this is sorted and it was all because the unit was on the bench and not in car.

More specifically:

1) Android Auto - Needs Canbus data to the head unit before it will function, without it the USB just detects and charges only. No idea why this is especially as CarPlay works fine on the bench.

2) Map updates - This one confused me at first as it didn't work in car; I'd only connected the main quad lock and display for testing, but then I thought perhaps the map update needs a valid GPS signal before it will update. That is indeed the case, the map update will only work with a valid GPS signal. Again very strange, perhaps it is ensuring the unit is in Europe before it allows a European map update.

Thanks again for all your help
Nick
